### ISSUE TITLE
Autodocumentation bug

### ISSUE THREAD

**Issue by Erhannis** [collerek/ormar#104]
**Describe the bug**
Not sure which library contains the bug - dunno if it's a problem with foreign keys, or with FastAPI, or something else.
When you have two classes X and Y with foreign keys to a class Z, if you have both an endpoint returning X and an endpoint returning Y, the autodocumentation (swagger and redoc) fails to generate, with the following error shown in the logs:
```
[32mINFO[0m:     127.0.0.1:56917 - "[1mGET /docs HTTP/1.1[0m" [32m200 OK[0m
[32mINFO[0m:     127.0.0.1:56917 - "[1mGET /openapi.json HTTP/1.1[0m" [91m500 Internal Server Error[0m
[31mERROR[0m:    Exception in ASGI application
Traceback (most recent call last):
  File "c:\users\USERNAME\appdata\local\programs\python\python38\lib\site-packages\uvicorn\protocols\http\h11_impl.py", line 394, in run_asgi
    result = await app(self.scope, self.receive, self.send)
  File "c:\users\USERNAME\appdata\local\programs\python\python38\lib\site-packages\uvicorn\middleware\proxy_headers.py", line 45, in __call__
    return await self.app(scope, receive, send)
  File "c:\users\USERNAME\appdata\local\programs\python\python38\lib\site-packages\fastapi\applications.py", line 199, in __call__
    await super().__call__(scope, receive, send)
  File "c:\users\USERNAME\appdata\local\programs\python\python38\lib\site-packages\starlette\applications.py", line 111, in __call__
    await self.middleware_stack(scope, receive, send)
  File "c:\users\USERNAME\appdata\local\programs\python\python38\lib\site-packages\starlette\middleware\errors.py", line 181, in __call__
    raise exc from None
  File "c:\users\USERNAME\appdata\local\programs\python\python38\lib\site-packages\starlette\middleware\errors.py", line 159, in __call__
    await self.app(scope, receive, _send)
  File "c:\users\USERNAME\appdata\local\programs\python\python38\lib\site-packages\starlette\exceptions.py", line 82, in __call__
    raise exc from None
  File "c:\users\USERNAME\appdata\local\programs\python\python38\lib\site-packages\starlette\exceptions.py", line 71, in __call__
    await self.app(scope, receive, sender)
  File "c:\users\USERNAME\appdata\local\programs\python\python38\lib\site-packages\starlette\routing.py", line 566, in __call__
    await route.handle(scope, receive, send)
  File "c:\users\USERNAME\appdata\local\programs\python\python38\lib\site-packages\starlette\routing.py", line 227, in handle
    await self.app(scope, receive, send)
  File "c:\users\USERNAME\appdata\local\programs\python\python38\lib\site-packages\starlette\routing.py", line 41, in app
    response = await func(request)
  File "c:\users\USERNAME\appdata\local\programs\python\python38\lib\site-packages\fastapi\applications.py", line 152, in openapi
    return JSONResponse(self.openapi())
  File "c:\users\USERNAME\appdata\local\programs\python\python38\lib\site-packages\fastapi\applications.py", line 130, in openapi
    self.openapi_schema = get_openapi(
  File "c:\users\USERNAME\appdata\local\programs\python\python38\lib\site-packages\fastapi\openapi\utils.py", line 354, in get_openapi
    definitions = get_model_definitions(
  File "c:\users\USERNAME\appdata\local\programs\python\python38\lib\site-packages\fastapi\utils.py", line 24, in get_model_definitions
    m_schema, m_definitions, m_nested_models = model_process_schema(
  File "pydantic\schema.py", line 525, in pydantic.schema.model_process_schema
  File "pydantic\schema.py", line 566, in pydantic.schema.model_type_schema
  File "pydantic\schema.py", line 222, in pydantic.schema.field_schema
  File "pydantic\schema.py", line 476, in pydantic.schema.field_type_schema
  File "pydantic\schema.py", line 751, in pydantic.schema.field_singleton_schema
  File "pydantic\schema.py", line 657, in pydantic.schema.field_singleton_sub_fields_schema
  File "pydantic\schema.py", line 476, in pydantic.schema.field_type_schema
  File "pydantic\schema.py", line 803, in pydantic.schema.field_singleton_schema
KeyError: <class 'pydantic.main.PkOnlyCA'>
```

**To Reproduce**
Run the following file with `uvicorn main:app --reload`:
```
# uvicorn main:app --reload
 
from fastapi import FastAPI
import databases
import ormar
from ormar import Model
import sqlalchemy
import pydantic
from typing import Optional, List
from uuid import UUID, uuid4
import asyncio
from enum import Enum
 
app = FastAPI()
DATABASE_URL = "sqlite:///db.sqlite"
database = databases.Database(DATABASE_URL)
metadata = sqlalchemy.MetaData()
 
class BaseMeta(ormar.ModelMeta):
    metadata = metadata
    database = database
 
class CA(ormar.Model):
    class Meta(BaseMeta):
        tablename = "cas"
    id: UUID = ormar.UUID(primary_key=True, default=uuid4)
    ca_name: str = ormar.Text(default="")
 
class CB1(ormar.Model):
    class Meta(BaseMeta):
        tablename = "cb1s"
    id: UUID = ormar.UUID(primary_key=True, default=uuid4)
    cb1_name: str = ormar.Text(default="")
    ca1: Optional[CA] = ormar.ForeignKey(CA, nullable=True)
 
class CB2(ormar.Model):
    class Meta(BaseMeta):
        tablename = "cb2s"
    id: UUID = ormar.UUID(primary_key=True, default=uuid4)
    cb2_name: str = ormar.Text(default="")
    ca2: Optional[CA] = ormar.ForeignKey(CA, nullable=True) 
 
#### DEBUGGING
# create the database
# note that in production you should use migrations
# note that this is not required if you connect to existing database
engine = sqlalchemy.create_engine(DATABASE_URL)
# just to be sure we clear the db before
metadata.drop_all(engine)
metadata.create_all(engine)
 
 
app = FastAPI()
app.state.database = database
 
@app.on_event("startup")
async def startup() -> None:
    database_ = app.state.database
    if not database_.is_connected:
        await database_.connect()
 
@app.on_event("shutdown")
async def shutdown() -> None:
    database_ = app.state.database
   if database_.is_connected:
        await database_.disconnect()
 
@app.get("/ca", response_model=CA)
async def get_ca():
    return None
 
@app.get("/cb1", response_model=CB1)
async def get_cb1():
    return None
 
@app.get("/cb2", response_model=CB2)
async def get_cb2():
    return None
```
Then navigate to http://localhost:8000/docs .  I believe an error screen will display.

Doing any of the following disrupts the bug:

1. Comment out one of the CB endpoints.
2. Remove either of the CB response_model annotation fields (e.g. delete `response_model=CB2`).
3. Remove either of the ForeignKeys to CA.

**Expected behavior**
Without a bug, I'd expect the documentation page to display normally.

**Screenshots**
~~If applicable, add screenshots to help explain your problem.~~

**Versions (please complete the following information):**
aiosqlite 0.16.1
SQLAlchemy 1.3.23
Python 3.8.0
ormar 0.9.3
pydantic 1.7.3
fastapi 0.63.0

**Additional context**
~~Add any other context about the problem here.~~

**Comment by collerek** [comment]
Fixed in 0.9.4, please update and check. :)

**Comment by Erhannis** [comment]
Confirmed fixed; thanks!
